src/main/java/uk/gov/cshr/service/InviteService.java

```diff
@@ -46,17 +47,29 @@ public Invite findByCode(String code) {
         return inviteRepository.findByCode(code);
     }
 
+    @ReadOnlyProperty
+    public Invite findByForEmail(String email) {
+        return inviteRepository.findByForEmail(email);
+    }
+
+    @ReadOnlyProperty
+    public Optional<Invite> findByForEmailAndStatus(String email, InviteStatus status) {
+        return inviteRepository.findByForEmailAndStatus(email, status);
+    }
+
     public boolean isCodeExpired(String code) {
         Invite invite = inviteRepository.findByCode(code);
+        return ifExpired(invite);
+    }
+
+    public boolean ifExpired(Invite invite) {
```

Comment:
Please rename this method to isCodeExpired.

Reply:
Done